base/interactiveutil.jl

```diff
@@ -466,9 +469,25 @@ end
     filename
 end
 
-function download(url::AbstractString)
-    filename = tempname()
-    download(url, filename)
+"""
+    download(url, [localfile]; silent=false)
+
+Download a file from the given url and save it to localfile, or to a temporary file if 
+localfile is ommited.
+Passing `silent=true` will suppress progress indicators.
+Note that this function relies on the availability of external tools such as `curl`, `wget`
+or `fetch` to download the file and is provided for convenience. For production use or
+situations in which more options are needed, please use a package that provides the desired
+functionality instead. 
+
+# Platform-Specific details
+
+On windows, a full URL must be provided, including the protocol prefix (e.g "http://" for
+HTTP)
+"""
+function download(url::AbstractString; silent::Bool=false)
```

Comment:
Documentation should be tied to the `download` function and not just this specific method.